x/xgit/repository.go

```diff
@@ -0,0 +1,41 @@
+package xgit
+
+import (
+	"errors"
+	"net/url"
+
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// Clone clones a repository from url into the path.
+func Clone(URL string, path string) error {
```

Comment:
This following line is still using `git.PlainClone`:
https://github.com/mesg-foundation/core/blob/8711a71037209288d39287805c7653f7bb043da9/mesg/deploy_deployer.go#L94

Reply:
Yes, and? What's about the PlainClone?

Reply:
Why not make the `deploy_deployer` file use the function from x packages?

Reply:
The following function has been moved to xgit:

https://github.com/mesg-foundation/core/blob/8711a71037209288d39287805c7653f7bb043da9/mesg/deploy_deployer.go#L80-L96

So let's remove it, and update the code to use the one from xgit

Reply:
Yes, but deploy_deplyer.go is inside another branch. So we need to merge this into the `dev` and then merge dev into that branch and update code.

Reply:
Done